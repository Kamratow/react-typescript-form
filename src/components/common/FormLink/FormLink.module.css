.FormLinkWrapper {
    margin-top: 15px;
    display: flex;
    justify-content: center;
}

.FormLinkDescription {
    font-weight: 600;
    display: block;
    margin-right: 10px;
}
