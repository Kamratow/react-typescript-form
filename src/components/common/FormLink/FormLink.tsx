import React from "react";
import { Link } from "react-router-dom";

import styles from "./FormLink.module.css";

export interface IFormLinkProps {
    description: string;
    linkText: string;
    linkPath: string;
}

const FormLink: React.FunctionComponent<IFormLinkProps> = ({
    description,
    linkText,
    linkPath,
}) => {
    return (
        <div className={styles.FormLinkWrapper}>
            <span className={styles.FormLinkDescription}>{description}</span>
            <div>
                <Link to={linkPath}>{linkText}</Link>
            </div>
        </div>
    );
};

export default FormLink;
