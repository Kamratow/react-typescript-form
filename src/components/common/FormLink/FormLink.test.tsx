import React from "react";
import { shallow } from "enzyme";

import FormLink, { IFormLinkProps } from "./FormLink";

const defaultProps: IFormLinkProps = {
    description: "test description",
    linkPath: "/testpath",
    linkText: "test link text",
};

describe("FormLink", () => {
    it("should render properly with default props", () => {
        const wrapper = shallow(<FormLink {...defaultProps} />);

        expect(wrapper).toMatchSnapshot();
    });
});
