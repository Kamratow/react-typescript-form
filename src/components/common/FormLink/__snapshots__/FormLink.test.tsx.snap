// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormLink should render properly with default props 1`] = `
<div
  className="FormLinkWrapper"
>
  <span
    className="FormLinkDescription"
  >
    test description
  </span>
  <div>
    <Link
      to="/testpath"
    >
      test link text
    </Link>
  </div>
</div>
`;
